### PR TITLE
[SPARK-14850] Show limit for array size when array is too big

### PR DESCRIPTION
**Pull request by tedyu**
## What changes were proposed in this pull request?

This PR shows the size of array and the limit when array is too big.
## How was this patch tested?

Existing tests.
@cloud-fan

### CODE REVIEW

File: sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java [hunk 1]
```diff
@@ -338,9 +338,10 @@ public UnsafeArrayData copy() {
   }
 
   public static UnsafeArrayData fromPrimitiveArray(int[] arr) {
-    if (arr.length > (Integer.MAX_VALUE - 4) / 8) {
+    int limit = (Integer.MAX_VALUE - 4) / 8;
+    if (arr.length > limit) {
       throw new UnsupportedOperationException("Cannot convert this array to unsafe format as " +
-        "it's too big.");
+        "its length (" + arr.length + ") exceeds " + limit + ".");
```
Comment (srowen) [inline sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java]: String interpolation?

**Comment by srowen** [comment]
Ted this is pretty trivial, and tacked on to another JIRA. It's not really a logically necessary part of the other changes. I don't think it needs a JIRA anyway; it's barely worth doing unless you're going to survey lots of similar messages and try to uniformly improve them. I'd rather not have to review a stream of tiny ad-hoc changes.

File: sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java (hunk 1 above)
Comment (tedyu) [inline sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java]: This is Java.

Comment (srowen) [inline sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeArrayData.java]: Oh, duh sorry about that.

**Comment by tedyu** [comment]
```
sbt.ForkMain$ForkError: java.lang.AssertionError: expected:<0.9986422261219262> but was:<0.9986422261219272>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.apache.spark.mllib.stat.JavaStatisticsSuite.testCorr(JavaStatisticsSuite.java:75)
```
The above assertion failure is not related to the change.

**Comment by cloud-fan** [comment]
This is really trivial, and will become unnecessary once we have new unsafe format for array. Can you close it for now and wait for the new format?

**Pull request closed without merging**